cranelift/filetests/filetests/isa/x86/simd-comparison-run.clif

```diff
@@ -1,6 +1,6 @@
 test run
 set enable_simd
-target x86_64
+target x86_64 haswell
```

Comment:
My understanding of this is that adding `haswell` here means that only hosts compatible with the `haswell` flags will run this test (that's what we documented in https://github.com/bytecodealliance/wasmtime/blob/master/cranelift/docs/testing.md#test-run). Is that the intent? Or do we want to try to run on `nehalem` and above since that has the SSE flags needed? (It's probably hard to find hardware to trigger this error...).

Reply:
This is actually reflecting a reality that's implicit at the moment: the host machine *must* match some SSE level to be able to effectively run the tests. I think this was implicit because the native host ISA would parse the CPUID, and all the machines running this would have the SSE SSE prerequisites. But this means that somebody who'd try to run the test on a very old machine might not be able to, and run into illegal instructions exceptions there.

This was uncovered by the change in this patch that uses the target ISA to compile, as defined in the test file, rather than the host's one built and configured from within the Rust code. (To answer your question in another comment, this was required to make it possible to pass the `use_new_backend` flag to the test.)

I'll identify what the precise SSE flags are, and will try to mitigate the potential SSE flags mismatch issue.

Reply:
It's actually quite complicated, because the target specific flags are not to be queried outside the TargetIsa impl. I tried a lot of different things and lost a lot of time, so I went back to something simpler (this is testing only, after all): ensure the host is Nehalem at least, and by policy ensure that no one will add run tests that require more than that. We'll need to find a more long-term solution that does the right thing, probably with a full refactoring of the flags system.